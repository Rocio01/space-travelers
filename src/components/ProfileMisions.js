import React from 'react';
import { useSelector } from 'react-redux';

const ProfileMisions = () => {
  const missionsStore = useSelector((store) => store.missionsReducer.missions);
  const joinedMissions = missionsStore.filter((mission) => mission.status === true);
  return (
    <ul className="list-group">
      {joinedMissions.map((mission) => (
        <li className="list-group-item" key={mission.misssion_id}>
          {mission.mission_name}
        </li>
      ))}
    </ul>
  );
};

export default ProfileMisions;
