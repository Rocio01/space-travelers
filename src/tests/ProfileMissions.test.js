import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import ProfileMisions from '../components/ProfileMisions';
import store from '../redux/configureStore';

test('matches ProfileMisions component snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <ProfileMisions />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
