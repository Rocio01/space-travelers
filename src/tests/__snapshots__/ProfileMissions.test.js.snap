// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches MissionsContainer component snapshot 1`] = `
<ul
  className="list-group"
/>
`;

exports[`matches ProfileMisions component snapshot 1`] = `
<ul
  className="list-group"
/>
`;
